dev/Kozani/Common/Logging.cpp

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation and Contributors.
+// Licensed under the MIT License.
+#include <pch.h>
+#include "Logging.h"
+
+void LogDebugMessage(PCSTR formatString, ...)
```

Comment:
wil does not recommend LOG_HR_MSG with S_OK as hr, although it allows it, with some warnings in runtime. So technically speaking, LOG_HR_MSG should not be used to log normal trace without failures. That said, we do use LOG_HR_MSG(S_OK, ...) in our windows codebase to achieve normal non-error case tracing. So I will replace it.

Reply:
WinAppSDK's logging mechanism is WIL, with a small sprinkling of Application eventlog. The latter can be improved for more natural (expansive) behavior (as we've desired) but there's some work to get there so for now just mark such things TODO and we'll supplement WIL logging in the future.

>wil does not recommend LOG_HR_MSG with S_OK as hr, although it allows it, with some warnings in runtime

Yes, which is why we shouldn't LOG_*() with S_OK. The added noise degrades the troubleshooting (and debugging) experience, and it's trivial enough to use appropos existing HRESULTs or if not invent new ones.

Reply:
Added new HRESULT code with FACILITY_ITF and use FACILITY_ITF with the new hresult for non-error logging.

// HRESULT used for logging non-error related information for tracing and debugging purpose with LOG_HR_MSG wil logging macro. 
// According to https://learn.microsoft.com/en-us/windows/win32/com/codes-in-facility-itf, all the COM-defined FACILITY_ITF codes have a code value 
// in the range of 0x0000-0x01FF. It is recommended that only code values in the range of 0x0200-0xFFFF be used for 3rd party FACILITY_ITF codes.
const HRESULT INFO_KOZANI{ MAKE_HRESULT(SEVERITY_SUCCESS, FACILITY_ITF, 0x0201) };

Reply:
NIT: Better to define it as `#define KOZANI_S_INFO _HRESULT_TYPEDEF_(0x00040201L)` - same way winerror.h defines them (_HRESULT_TYPEF_) and named similarly (FOO_sev_NAME e.g. RPC_S_BLAH for a successful RPC return code).

If you want to quibble on the preprocessor then shouldn't this be `constexpr` rather than `const` ?

Reply:
Used constexpr and added comments with the final HRESULT error code in both HEX and DEC forms for better searching:

// Success HRESULT used for Kozani information logging purpose.
// 0x00040201L (262657)
constexpr HRESULT KOZANI_S_INFO{ MAKE_HRESULT(SEVERITY_SUCCESS, FACILITY_ITF, 0x0201) };